### DISCUSSION TITLE
General questions about OpenDeck posibilities

### DISCUSSION THREAD

**Discussion by d27admino** [shanteacontrols/OpenDeck#55] [General]
Hello,
I just found your website. It looks cool.
I am a beginner builder and would like to have confirmation about what you can do or not do with open deck.

Sorry for the format that looks like a exam, but it's to make it easier to read/reply!.

so here are some questions; 

1 - Is Open Deck 100%  standalone? 
Do I just need to solder/connect faders and knobs to the deck and it will work ? 

2 - If I use faders, they take the place of a potentiometer? or of an encoder ?  
This is for planning the number of potis/faders/buttons that i can have . 

3 - When the page says "64 digital inputs for 64 buttons or 32 encoders
32 analog inputs which can be used as button inputs as well"

Can potentiometers and faders be connected on the 32 encoders too? or only the 32 analog inputs ? 
FOr example, could I have a box with 64 faders?or only 32 ? :-) (it is just for the example to understand the possible connections).

4- Faders should on ly be classic faders. Not motorized, and not "fader with led" neither. Right ? 

5 - What can the leds be used for ? 
I understand that is mainly to show midi activity for 1 midi channel, but will not show values, and will not fade light intensity according to midi value. So 16 leds would be the maxium to be used (eventhough less is more realistic, to show midi activity on a few midi channels). 

6- What are the extra controllers be used for ? 
"SPI connector " , "STM32F405 MCU capable of hosting 10 presets" , "Arduino" etc. (I am a noob for this).
Why would they be used and how? Are they "necessary" for open deck to work or to have some functions (presets?) ? 

7 - Additional UART connector for touchscreens - Additional I2C connector for small LCD/OLED displays
Where can i find more info about the possibilities and how to use them ? 
For example, can i use the screens to display potentiometer name or value? or is it for something else ? 

8 - When you say it takes you 3 months to make 1 of your controller, is it because of the time you have or because there are parts that take time? experience would help. 

Thanks a lot if you have the time to answer the questions. It would help me see clearer.

**Comment by paradajz** [comment]
Hi,

glad you're interested in the project. Here goes:

1) I'm not sure what you're referring to. Buttons on OpenDeck work straight away, as for the analog components, you need to explicitly enable them in the web configurator tool. You don't need to install any additional software or drivers - OpenDeck is compatible with any DAW on any OS.

2) Faders and potentiometers are technically the same thing - they are just variable resistance components (or analog components). Encoder is digital component since it outputs quadrature signal and is therefore connected to digital inputs. This is documented in the wiki.

3) 64 digital inputs for 64 buttons or 32 encoders means that there really are 64 digital inputs, so you can use that to connect 64 buttons. Encoders, however, take up 2 digital input pins, so, if you were to connect 32 encoders to the OpenDeck board, you would need all 64 digital inputs on the board. There are also 32 analog inputs on the board, and you can use them to connect either analog components (faders, potentiometers, FSR etc.) or you can configure analog input in the web configurator tool to act as an digital input, so you can connect buttons on those inputs. You can't connect encoders to analog inputs, however, since they require faster readout than normal readout on analog inputs.

4) Motorized faders aren't supported. I'm not sure what you're refering to by "fader with LED". If it's just a normal LED then you can connect it to digital output on the board.

5) LEDs are generally used as indicators of some state of your software. In total you can connect 48 LEDs to OpenDeck board. Using the web configurator you decide what turns them on - MIDI channel, MIDI value, MIDI velocity, MIDI message, MIDI input or local control etc. 

6/7) SPI currently isn't used for anything. The plan is to support SPI sensors in the future. I2C connector is currently used for small OLED/LCD displays. UART connector is used for touchscreens such as Nextion.

8) Both because of the time I have and waiting times for other components.

Almost everything you asked is documented in the Wiki so please go through that.
https://github.com/paradajz/OpenDeck/wiki

**Reply by d27admino** [reply]
Hi, Thanks a lot for the reply and confirmations. SOmetimes, as I am not very technical so I don't necessarily understand written instruction (or need a confirmation that I understood). 

Anyway, for your question about "faders with leds", they are the same you can find on some synths and eurorack modules (like behringer used them in their arp clones).   (example:   https://fr.aliexpress.com/item/32819462017.html?spm=a2g0o.search0306.0.0.5b3c56abxUzAnN&algo_pvid=dce9adfa-2034-4d3e-a6dc-c388d43ac7e4&algo_expid=dce9adfa-2034-4d3e-a6dc-c388d43ac7e4-3&btsid=2100bdd716178701437323727ea428&ws_ab_test=searchweb0_0,searchweb201602_,searchweb201603_)  )

it is not a led that is connected to a modulator, it switches on when their is power. They can be useful for a multi-band equalizer (they are not good for mixer levels because more fragile and less esay to grab and move).

**Comment by d27admino** [comment]
>>> I'm not sure what you're referring to. Buttons on OpenDeck work straight away, as for the analog components, you need to explicitly enable them in the web configurator tool. You don't need to install any additional software or drivers - OpenDeck is compatible with any DAW on any OS.

I think you answered. The question was to know if I need arduino or similar components in adition to OpenDeck to be useable.

I understand that I do not need Arduino. I just need to properly solder faders, knobs, button to open deck, configure them, and it will all be functional. straight away. I do not even need midi LoopBe or other software to recognize the midi.
So it means I can use my midi device based on open deck without a PC (directly connected to synths/etc
Am I right ?

**Reply by paradajz** [reply]
> I do not even need midi LoopBe or other software to recognize the midi.

No, OpenDeck uses native, class-compliant USB MIDI.

> So it means I can use my midi device based on open deck without a PC (directly connected to synths/etc

Yes.

**Comment by AlpesMachines** [comment]
thanks for all those explanations. 

is it possible to implement an application inside the Opendeck to send specific sysex or control change messages (+ notes on/off) to a hardware synth ? store its patches ? and bonus : code an arpeggiator/sequencer ? (a bit like the midibox.org and MIOS/app)

**Reply by paradajz** [reply]
>send specific sysex

Not yet but this is planned.

>or control change messages

You can do this already.

>store its patches

Not sure what you mean with this.

>code an arpeggiator/sequencer ? (a bit like the midibox.org and MIOS/app)

No, this is out of scope for OpenDeck.

**Comment by AlpesMachines** [comment]
> store its patches
hardware synths usually store a tone parameters (and name !) using a sysex, this sysex corresponding to the parameters of osc, filter, vca, env etc
sending a program change to the Opendeck would make open deck check in its memory for a patch, concatenate the parameters into a sysex and send this sysex to the synth

more question : is it possible to have easily several USART DIN Midi ? by hacking your code
do you plan to support liquid crystal display ?

**Reply by paradajz** [reply]
LCD displays and touchscreens are supported. Multiple DINs aren't supported nor am I planning to for now at least.

**Reply by AlpesMachines** [reply]
characters display ? like the HD44780 16colomns x2rows costing 1€99

**Reply by paradajz** [reply]
https://github.com/paradajz/OpenDeck/wiki/Connections

**Comment by kaziorkaski** [comment]
Is it possible to add features in the settings editor so that encoders and potentiometers work in **Relative mode**. Currently, I have the impression that they work only in **absolute mode**. This gives a very inelegant, _jumping_ behavior of the knob. Such a solution can be found, for example, in Artuia Polybrute.  😊
Greetings 👋🏻

**Reply by kaziorkaski** [reply]
My early prototype is starting to work 😎![IMG_4089](https://github.com/user-attachments/assets/07b49745-c6d4-4456-aff6-0d8f90595997)
![IMG_4088](https://github.com/user-attachments/assets/8db4758f-9734-4635-bb90-a495a0c88e68)

![IMG_4087](https://github.com/user-attachments/assets/302d8138-6636-481e-910b-cd6ada84f8ef)

**Reply by paradajz** [reply]
> Is it possible to add features in the settings editor so that encoders and potentiometers work in **Relative mode**. Currently, I have the impression that they work only in **absolute mode**. This gives a very inelegant, _jumping_ behavior of the knob. Such a solution can be found, for example, in Artuia Polybrute. 😊 Greetings 👋🏻

I'm not sure what you're referring to.
